src/main/kotlin/com/rooplor/classcraftbackend/services/AuthService.kt

```diff
@@ -0,0 +1,17 @@
+package com.rooplor.classcraftbackend.services
+
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseAuthException
+import com.google.firebase.auth.FirebaseToken
+import org.springframework.stereotype.Service
+
+@Service
+class AuthService {
+    fun validateIdToken(idToken: String): FirebaseToken? =
+        try {
+            val decodedToken = FirebaseAuth.getInstance().verifyIdToken(idToken)
+            decodedToken
+        } catch (e: FirebaseAuthException) {
+            null
+        }
+}
```

Comment:
Could help to add unit-test for check function can catch correct error exception?

Reply:
for this one cannot create a unit-test because it require valid `serviceAccountKey.json` cannot use invalid key. in local it possible but in github ci it will fail.